tests/TestCase/Date/DiffTest.php

```diff
@@ -0,0 +1,375 @@
+    {
+        $dt = ChronosDate::createFromDate(2000, 1, 1);
+        $this->assertSame(1, $dt->diffInYears($dt->subYears(1)));
+    }
+
+    public function testDiffInYearsVsDefaultNow()
+    {
+        $this->wrapWithTestNow(function () {
+            $this->assertSame(1, ChronosDate::now()->subYears(1)->diffInYears());
+        });
+    }
+
+    public function testDiffInYearsEnsureIsTruncated()
+    {
+        $dt = ChronosDate::createFromDate(2000, 1, 1);
+        $this->assertSame(1, $dt->diffInYears($dt->addYears(1)->addMonths(7)));
+    }
+
+    public function testDiffInMonthsPositive()
+    {
+        $dt = ChronosDate::createFromDate(2000, 1, 1);
+        $this->assertSame(13, $dt->diffInMonths($dt->addYears(1)->addMonths(1)));
+    }
+
+    public function testDiffInMonthsNegativeWithSign()
+    {
+        $dt = ChronosDate::createFromDate(2000, 1, 1);
+        $this->assertSame(-11, $dt->diffInMonths($dt->subYears(1)->addMonths(1), false));
+    }
+
+    public function testDiffInMonthsNegativeNoSign()
+    {
+        $dt = ChronosDate::createFromDate(2000, 1, 1);
+        $this->assertSame(11, $dt->diffInMonths($dt->subYears(1)->addMonths(1)));
+    }
+
+    public function testDiffInMonthsVsDefaultNow()
+    {
+        $this->wrapWithTestNow(function () {
+            $this->assertSame(12, ChronosDate::now()->subYears(1)->diffInMonths());
+        });
+    }
+
+    public function testDiffInMonthsEnsureIsTruncated()
+    {
+        $dt = ChronosDate::createFromDate(2000, 1, 1);
+        $this->assertSame(1, $dt->diffInMonths($dt->addMonths(1)->addDays(16)));
+    }
+
+    public function testDiffInDaysPositive()
+    {
+        $dt = ChronosDate::createFromDate(2000, 1, 1);
+        $this->assertSame(366, $dt->diffInDays($dt->addYears(1)));
+    }
+
+    public function testDiffInDaysNegativeWithSign()
+    {
+        $dt = ChronosDate::createFromDate(2000, 1, 1);
+        $this->assertSame(-365, $dt->diffInDays($dt->subYears(1), false));
+    }
+
+    public function testDiffInDaysNegativeNoSign()
+    {
+        $dt = ChronosDate::createFromDate(2000, 1, 1);
+        $this->assertSame(365, $dt->diffInDays($dt->subYears(1)));
+    }
+
+    public function testDiffInDaysVsDefaultNow()
+    {
+        $this->wrapWithTestNow(function () {
+            $this->assertSame(7, ChronosDate::now()->subWeeks(1)->diffInDays());
+        });
+    }
+
+    public function testDiffInDaysEnsureIsTruncated()
+    {
+        $dt = ChronosDate::createFromDate(2000, 1, 1);
+        $this->assertSame(1, $dt->diffInDays($dt->addDays(1)->addHours(13)));
+    }
+
+    /* Does not include January 31 since DatePeriod does not include the end date by default
+    public function testDiffInDaysFilteredPositiveWithMutated()
+    {
+        $dt = ChronosDate::createFromDate(2000, 1, 1);
+        $this->assertSame(5, $dt->diffInDaysFiltered(function ($date) {
+            return $date->dayOfWeek === 1;
+        }, $dt->endOfMonth()));
+    }
+    */
```

Comment:
This test just happens to pass with Chronos because we use the current time instead of setting to 00:00:00. DatePeriod does not include the end date and the option to enable that was added in PHP 8.2.

Reply:
Think I can work around this in 8.1 by adding time to the end date.

Reply:
Having date operations behave differently based on the time of day they are run is unexpected. Date operations should behave like calendar dates do.

I'm happy to do a follow up pull request to improve how this works.